src/compiler/compile/css/Stylesheet.ts

```diff
@@ -289,14 +293,32 @@ export default class Stylesheet {
 
 	nodes_with_css_class: Set<CssNode> = new Set();
 
-	constructor(source: string, ast: Ast, filename: string, dev: boolean) {
+	constructor({
+		source,
+		ast,
+		component_name,
+		filename,
+		dev,
+		get_css_hash = get_default_css_hash
+	}: {
+		source: string;
+		ast: Ast;
+		filename: string | undefined;
+		component_name: string | undefined;
+		dev: boolean;
+		get_css_hash: CssHashGetter;
+	}) {
 		this.source = source;
 		this.ast = ast;
 		this.filename = filename;
 		this.dev = dev;
 
 		if (ast.css && ast.css.children.length) {
-			this.id = `svelte-${hash(ast.css.content.styles)}`;
+			this.id = get_css_hash({
```

Comment:
Maybe we should add a [saftynet](https://github.com/sveltejs/svelte/pull/4740/files#diff-f99033a09b5f0a1bdc31b7ca72b308f4ad1cb04d5219ca8f96eaac124971c9ee) here and warn the user if he is passing in a hash that is none compliant (similar to a11y checks).